### PR TITLE
chore(Model): API and DataStore integration test for scalars, lists, enums

### PR DESCRIPTION
**Pull request by lawmicha**
*Issue #, if available:*
https://github.com/aws-amplify/amplify-ios/issues/314

*Description of changes:*
This adds integration tests for some models containing scalars, lists, lists with enums, etc.

*Check points:*

- [x] Added new tests to cover change, if needed
- [x] All unit tests pass
- [x] All integration tests pass
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLModelBased/GraphQLScalarTests.swift [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            stringList: ["value1"],
+            stringNullableList: [],
+            nullableStringList: [],
+            nullableStringNullableList: nil)
+
+        let updatedContainer = ListStringContainer(id: container.id,
+                                                   test: "test",
+                                                   nullableString: "test",
+                                                   stringList: ["value1"],
+                                                   stringNullableList: ["value1"],
```
Comment (diegocstn) [inline AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLModelBased/GraphQLScalarTests.swift:116]: should we also test when these hold `nil` if/where possible?

Comment (lawmicha) [reply AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLModelBased/GraphQLScalarTests.swift:152]: yes that sounds good, adding `testListContainerWithNil` test for APi and DataStore

**Review by lawmicha — COMMENTED** [review]
splitting the asserts into individual CRUD operations to make it easier to triage test failures

**Review by diegocstn — APPROVED** [review]
